### PR TITLE
Apps: fixed autocomplete with no fields

### PR DESCRIPTION
**Pull request by levb**
I was facing numerous exceptions adding a `/apps info` command that had no form, or even a form with no fields. This PR should fix it.

@mickmister I had to disable a couple tests, we should discuss the parser improvements offline, they should address the (in-)significance of trailing space.

### CODE REVIEW

**Review by larkox — APPROVED** [review]
LGTM. @mickmister any of these changes have to be reflected on the mobile PR?

File: components/suggestion/command_provider/app_command_parser.ts [hunk 1]
```diff
@@ -652,47 +656,43 @@ export class AppCommandParser {
...
             return null;
         }
+        bindings = root.bindings;
 
-        const searchThroughBindings = (binding: AppBinding, pretext: string): AppBinding => {
-            if (!binding.bindings || binding.bindings.length === 0) {
-                return binding;
+        for (const t of tokens) {
+            let matched: AppBinding| null = null;
+            for (const b of bindings) {
```
Comment (mickmister) [inline components/suggestion/command_provider/app_command_parser.ts:680]: Does this non-recursive approach work for a deeply nested structure?

**Review by mickmister — APPROVED** [review]
Looks good :+1: A few comments/questions and one non-blocking suggestion

**Review by mickmister — COMMENTED** [review]
Comments from video call

**Comment by levb** [comment]
Replacing with https://github.com/mattermost/mattermost-webapp/pull/7426

**Pull request closed without merging**